pkg/hooks/audit_logger_sqlite.go

```diff
@@ -0,0 +1,137 @@
+package hooks
+
+import (
+	"database/sql"
+
+	_ "github.com/mattn/go-sqlite3"
+	"github.com/square/p2/pkg/logging"
+	"github.com/square/p2/pkg/util"
+)
+
+type SQLiteAuditLogger struct {
+	sqlite *sql.DB
+	logger *logging.Logger
+}
+
+func NewSQLiteAuditLogger(sqliteDBPath string, logger *logging.Logger) (*SQLiteAuditLogger, error) {
+	db, err := sql.Open("sqlite3", sqliteDBPath)
+	if err != nil {
+		return nil, err
+	}
+	if db == nil {
+		return nil, util.Errorf("Unable to create a sqlite connection.")
+	}
+	al := &SQLiteAuditLogger{
+		sqlite: db,
+		logger: logger,
+	}
+	if err := al.ensureMigrated(); err != nil {
+		return nil, err
+	}
+	return al, nil
+}
+
+func (al *SQLiteAuditLogger) log(ctx *HookExecContext, success bool) {
+	env := ctx.env
+	stmt := `INSERT INTO hook_results(
+pod_id,
+pod_unique_key,
+hook_name,
+hook_stage,
+success
+) VALUES(?,?,?,?,?)`
+	podID := env.HookedPodIDEnvVar
+	podUniqueKey := env.HookedPodUniqueKeyEnvVar
+	hookName := ctx.Name
+	hookStage := env.HookEventEnvVar
+	// Boolean values are represented by TINYINTs in SQL. Adapt.
+	dbSuccess := 0
+	if !success {
+		dbSuccess = 1
+	}
+
+	_, err := al.sqlite.Exec(stmt, podID, podUniqueKey, hookName, hookStage, dbSuccess)
```

Comment:
https://www.sqlite.org/faq.html#q3
> (5) When any process wants to write, it must lock the entire database file for the duration of its update. But that normally only takes a few milliseconds. Other processes just wait on the writer to finish then continue about their business. 

So it claims to work as I expect. Given that you've worked with sqlite in production before, maybe you have some more information that I don't.